lm_eval/evaluator.py

```diff
@@ -129,25 +133,30 @@ def simple_evaluate(
             if task_obj is None:
                 continue
 
-        config = task_obj._config
-        if config["output_type"] == "generate_until" and gen_kwargs is not None:
-            config["generation_kwargs"].update(gen_kwargs)
+        if task_obj.get_config("output_type") == "generate_until":
+            if gen_kwargs is not None:
+                task_obj.override_config(
+                    key="generation_kwargs", value=gen_kwargs, update=True
+                )
+
+            if predict_only:
+                log_samples = True
+                eval_logger.info(
+                    f"Processing {task_name} in output-only mode. Metrics will not be calculated!"
+                )
+                # we have to change the class properties post-hoc. This is pretty hacky.
+                task_obj.override_metric(metric_name="bypass")
```

Comment:
Seems like source for fails too, when task is an instance of Task with no override_metric available. Did not find any PR addressing this. May be missing something.

Reply:
@baberabb could you look into this?

Reply:
@LSinev thanks for catching this! I'll just move this to `Task`